src/f32/funcs.rs

```diff
@@ -85,6 +92,28 @@ pub(crate) mod sse2 {
     // _ps_const_ty!(PS_COSCOF_P2, f32x4, 4.166_664_6e-2);
     // _ps_const_ty!(PS_CEPHES_FOPI, f32x4, 1.273_239_5); // 4 / M_PI
 
+    _ps_const_ty!(g_XMNegativeZero, u32x4, 0x80000000);
+    _ps_const_ty!(g_XMPi, f32x4, std::f32::consts::PI);
+    _ps_const_ty!(g_XMHalfPi, f32x4, std::f32::consts::FRAC_PI_2);
+    _ps_const_ty!(g_XMSinCoefficients0 , f32x4, -0.16666667, 0.0083333310, -0.00019840874, 2.7525562e-06);
+    _ps_const_ty!(g_XMSinCoefficients1 , f32x4, -2.3889859e-08, -0.16665852 /*Est1*/, 0.0083139502 /*Est2*/, -0.00018524670 /*Est3*/);
+    _ps_const_ty!(g_XMOne, f32x4, 1.0);
+    _ps_const_ty!(g_XMTwoPi, f32x4, std::f32::consts::PI * 2.0);
+    _ps_const_ty!(g_XMReciprocalTwoPi , f32x4, 0.159154943);
+
+    macro_rules! XM_PERMUTE_PS {
```

Comment:
I'm not sure it's worth keeping these macros. I think they can be useful when dealing with different implementations of common operations, (e.g. SSE2 and SSE4) but so far I've just stuck with SSE2 and written everything in explicit intrinsices.

Reply:
The `fma` instructions seem to add a significant speedup. 

```
+avx ~5%
+fma ~20%
+fma,+avx ~20%
```

Reply:
Hmm, I seem to get different results when running all of the quat tests vs just the quat slerp test. The `fma` instruction is without question an improvement. It's not clear whether the `avx` one is useful or not.

Reply:
Not sure what you mean, the quat slerp test is changing the result of the other tests? Or enabling AVX is changing results? Having FMA off or on could definitely give different results.

Reply:
With `XM_PERMUTE_PS` and `XM_FMADD_PS` I'd prefer to make them inline functions and name them `m128_permute` and `m128_mul_add`. Note that `avx` and `fma` features aren't enabled by default in x64 builds and I probably won't add build permutations for these to the CI, just because it already takes an hour as is so they won't be getting tested. It might be worth making a note about that in the README. Well, if I could add a travis build that wasn't waited on by the CI (so PR's weren't waiting on them) but still let me know if they are broken maybe I could add an automated test.